### ISSUE TITLE
F1 2012 [BLUS31014]: white screen only

### ISSUE THREAD

**Issue by TotalCaesar659** [RPCS3/rpcs3#4999]
In https://github.com/RPCS3/rpcs3/pull/4973, F1 2012 has white screen only and many warnings

> ·E 0:00:35.316872 {rsx::thread} RSX: Invalid FIFO queue get/put registers found, get=0x1000, put=0x94A4
> ·E 0:00:35.316875 {rsx::thread} RSX: Application has failed to recover, resetting FIFO queue

that cause memory leak.
[F1_2012.zip](https://github.com/RPCS3/rpcs3/files/2299608/F1_2012.zip)

**Comment by kd-11** [comment]
If this is a race condition introduced by the speed boost in that PR, then it is a duplicate of https://github.com/RPCS3/rpcs3/issues/2359. If not, could be introduced by the previous commits changing memory addressing. The AAA fixes PR by itself does not modify vm memory behaviour in any way.
To determine if its a race condition, check if the issue is always present for everyone. You will likely find someone who can get ingame. Also check if HLE gcm fixes the problem (select manual load library and load the necessary libraries only (spurs_jq, sre, resc, rtc)

**Comment by TotalCaesar659** [comment]
> check if the issue is always present for everyone
I don't understand, do you want me to test previous builds before AAA fixes?
> check if HLE gcm fixes the problem
OK, I'll do it now.

**Comment by MSuih** [comment]
>I don't understand, do you want me to test previous builds before AAA fixes?

No, he wants you to find someone else who owns that game and check if they can't get ingame either. If it's a race condition then some hardware combinations can probably get it ingame.

But since you are already blaming #4973 for the regression then you're implying that it worked before that PR. If you haven't tested any other builds then you should do that before accusing KD's changes for the regression.

**Comment by TotalCaesar659** [comment]
> No, he wants you to find someone else who owns that game and check if they can't get ingame either. If it's a race condition then some hardware combinations can probably get it ingame.

Ah, I got this. Then maybe somebody else will see this thread and test it.

> If you haven't tested any other builds then you should do that before accusing KD's changes for the regression.

If I blame some PR for bug, then it hadn't exist before this PR. I've tested this game on almost every PR. So If you think that I don't have enough for this, you're mistaking.

> check if HLE gcm fixes the problem (select manual load library and load the necessary libraries only (spurs_jq, sre, resc, rtc)

Done, it works now.
![f1_2012](https://user-images.githubusercontent.com/14265316/44300606-920f1180-a312-11e8-8fd5-a516dc8fd865.png)
[F1_2012.zip](https://github.com/RPCS3/rpcs3/files/2299641/F1_2012.zip)

**Comment by MSuih** [comment]
Oh, it was just a misunderstanding then. I apologize for the tone of my message.

**Comment by kd-11** [comment]
If HLE gcm fixes it then it suffers from the same gcm bug as skate3, which again, is not something that the PR touches. There were several vm memory patches committed into master in the last few days, check if it can be narrowed down to one of them. If not then its likely a race condition of some kind.

**Comment by TotalCaesar659** [comment]
Finally, I've found the commit: the game works with https://github.com/RPCS3/rpcs3/pull/4953 and doesn't do with https://github.com/RPCS3/rpcs3/pull/4975.
[F1_2012.zip](https://github.com/RPCS3/rpcs3/files/2299911/F1_2012.zip)

**Comment by AniLeo** [comment]
Retest with latest master, #5002 could have fixed this

**Comment by TotalCaesar659** [comment]
Unfortunately, same result in https://github.com/RPCS3/rpcs3/pull/5010.
[F1_2012.zip](https://github.com/RPCS3/rpcs3/files/2300860/F1_2012.zip)
